bytearray_operations.go

```diff
@@ -44,6 +44,83 @@ func LeftShift(data []byte, shift uint64) []byte {
 	return result
 }
 
+// Mask apply AND mask to byte array
+func Mask(data, mask []byte) []byte {
+	var dataLength = len(data)
+	result := make([]byte, dataLength)
+	if dataLength < 1 {
+		return result
+	}
+	copy(result, data)
+
+	var maskLength = len(mask)
+	var operationLength = maskLength
+	// If mask is longer than data, keep operation to data length
+	if maskLength > dataLength {
```

Comment:
Can be reduce to 
```go
var operationLength = len(mask)
if operationLength > dataLength {
```

Reply:
Its actually there for readability. I can reduce if you prefer.

Reply:
Ok, you can keep it for readability